core/src/main/resources/static/js/envs.js

```diff
@@ -662,44 +662,65 @@ app.controller("envsCtrl", function($scope, $http, $location, $window) {
                         else if($scope.addNewCluster.type === 'kafkaconnect')
                             clusterType = 'KAFKA_CONNECT';
 
-                        var serviceInput = {};
+                        if($scope.addNewCluster.protocol === "PLAINTEXT"){
+                            swal({
+                                title: "PLAINTEXT protocol is unsecure !",
+                                text: "Would you like to proceed anyways ?",
+                                type: "warning",
+                                showCancelButton: true,
+                                confirmButtonColor: "#DD6B55",
+                                confirmButtonText: "Yes !",
+                                cancelButtonText: "No, cancel please!",
+                                closeOnConfirm: true,
+                                closeOnCancel: true,
+                                timer: 10000
+                            }).then(function(isConfirm) {
+                                if (isConfirm.value) {
+                                    var serviceInput = {};
+
+                                    serviceInput['clusterName'] = $scope.addNewCluster.envname;
+                                    serviceInput['bootstrapServers'] = $scope.addNewCluster.host;
+                                    serviceInput['protocol'] = $scope.addNewCluster.protocol;
+                                    serviceInput['clusterType'] = clusterType;
+                                    serviceInput['projectName'] = $scope.addNewCluster.projectName;
+                                    serviceInput['serviceName'] = $scope.addNewCluster.serviceName;
+                                    serviceInput['kafkaFlavor'] = $scope.kafkaFlavor;
+                                    serviceInput['associatedServers'] = $scope.addNewCluster.associatedServers;
+
+                                    $http({
+                                        method: "POST",
+                                        url: "addNewCluster",
+                                        headers : { 'Content-Type' : 'application/json' },
+                                        params: {'addNewCluster' : serviceInput },
```

Comment:
are we intentionally sending both `params` and `data` (which contain the same thing) within this request?

Reply:
Updated. Removed params